File: MagicHax/android/app/src/main/java/me/nathanp/magiccreatures/viewmodel/CreatureListViewModel.java
package me.nathanp.magiccreatures.viewmodel;

import android.app.Application;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import me.nathanp.magiccreatures.model.Creature;
import me.nathanp.magiccreatures.model.CreatureRepository;

public class CreatureListViewModel extends AndroidViewModel {
    private static final String TAG = "CreatureListViewModel";
    private CreatureRepository creatureRepository;

    public CreatureListViewModel(@NonNull Application application) {
        super(application);
        creatureRepository = new CreatureRepository();
    }

    public void add(Creature creature) {
        creatureRepository.addCreature(creature);
    }

    public void update(String which, Creature creature) {
        creatureRepository.addCreature(creature);
    }

    public void delete(String which) {
        creatureRepository.deleteCreature(which);
    }

    public LiveData<List<Creature>> getAllCreatures() {
        return creatureRepository.getAllCreatures();
    }
}

File: MagicHax/android/app/src/main/java/me/nathanp/magiccreatures/viewmodel/CreatureListViewModel.kt
package me.nathanp.magiccreatures.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import me.nathanp.magiccreatures.model.Creature
import me.nathanp.magiccreatures.model.CreatureRepository

class CreatureListViewModel(application: Application) : AndroidViewModel(application) {
    private val creatureRepository = CreatureRepository()
    fun add(creature: Creature) {
        creatureRepository.addCreature(creature)
    }

    fun update(which: String, creature: Creature) {
        creatureRepository.addCreature(creature)
    }

    fun delete(which: String) {
        creatureRepository.deleteCreature(which)
    }

    val allCreatures: LiveData<List<Creature>>
        get() = creatureRepository.allCreatures

    companion object {
        private const val TAG = "CreatureListViewModel"
    }
}